source/extensions/quic_listeners/quiche/active_quic_listener.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "envoy/api/v2/listener/quic_config.pb.h"
+#include "envoy/network/connection_handler.h"
+#include "envoy/network/listener.h"
+
+#include "server/connection_handler_impl.h"
+
+#include "extensions/quic_listeners/quiche/envoy_quic_dispatcher.h"
+
+namespace Envoy {
+namespace Quic {
+
+// QUIC specific UdpListenerCallbacks implemention which delegates incoming
+// packets, write signals and listener errors to QuicDispatcher.
+class ActiveQuicListener : public Network::UdpListenerCallbacks,
+                           public Server::ConnectionHandlerImpl::ActiveListenerImplBase,
+                           // Inherits below two interfaces just to have common
```

Comment:
Can you add a TODO to potentially clean this up? The level of dynamic casting, etc. that we have now in the connection handling path does not thrill me. I think we can do better here but it will take some thought on what we need to do.

Reply:
done